src/main/java/calculator/Formula.java

```diff
@@ -0,0 +1,57 @@
+package calculator;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class Formula {
+    private Queue<FormulaElement> formula;
+
+    public Formula(String[] inputs) {
+        validateSizeOfInputsIsOdds(inputs);
+        formula = new LinkedList<>();
+        generateFormula(inputs);
+    }
+
+    public double calculateFormula() {
+        Operand operand1 = (Operand) formula.poll();
+        double result = operand1.getOperand();
+
+        while (formula.size() > 0) {
+            Operator operator = (Operator) formula.poll();
+            Operand operand2 = (Operand) formula.poll();
+            result = operator.calculate(operand1, operand2);
+            operand1 = new Operand(result);
+        }
+        return result;
+    }
+
+    private void generateFormula(String[] inputs) {
+        for (String input : inputs) {
+            formula.offer(generateFormulaElement(input));
+        }
+    }
+
+    private FormulaElement generateFormulaElement(String input) {
+        if (isOperandTurn()) {
+            return new Operand(input);
+        }
+        return new Operator(input);
+    }
+
+    private boolean isOperandTurn() {
+        return !isOddNumber(formula.size());
```

Comment:
```suggestion
        return isEvenNumber(formula.size());
```

!와 같은 부정문은 가독성을 해칠 수 있어요.